### PR TITLE
Change attaching-files to rst

### PR DESCRIPTION
**Pull request by lfbrock**
Switching the attaching files document to rst.

### CODE REVIEW

**Comment by lfbrock** [comment]
Pandoc seems to use `~~~`, should we switch the style guide to make it more convenient when converting documents? 

@esethna @JeffSchering any thoughts?

**Comment by JeffSchering** [comment]
That seems reasonable. If there are no objections, I'll create a PR to change the style guide from `^^^^` to `~~~~~`

**Comment by esethna** [comment]
Thanks @JeffSchering, makes sense to update the guide

File: source/help/messaging/attaching-files.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
+Attaching Files
```
Comment (esethna) [inline source/help/messaging/attaching-files.rst]: The index file for this guide section needs to be updated when the file name changes:

https://github.com/mattermost/docs/blob/master/source/guides/user.rst

**Comment by lfbrock** [comment]
@esethna updated the file name, please review?

File: source/help/messaging/attaching-files.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Attachment Methods
+------------------
+Attach a file by drag and drop or by clicking the attachment icon in the
+message input box.
+
+Drag and Drop
+~~~~~~~~~~~~~
+Upload a file or selection of files by dragging the files from your
+computer into the right-hand sidebar or center pane. Dragging and
+dropping attaches the files to the message input box, then you can
```
Comment (yangchen1) [inline source/help/messaging/attaching-files.rst]: Feels like a run-on sentence to me. Hard to-read, according to Hemmingway app. Propose: "Dragging and dropping attaches the files to the message input box. Then, you can..."

**Pull request merged**